### ISSUE TITLE
Feature request: Customize OPTIONS return code

### ISSUE THREAD

**Issue by avaliente-evs** [CrowCpp/Crow#939]
Hello,

I'm working on a server application whose existing (external) clients expects to have 200 on OPTIONS, and I cannot do that currently with crow.
Would it be possible to customize the return code for OPTIONS requests?

I can write the feature and send a PR, but I would like to know if this behavior is OK for you?

**Comment by gittiver** [comment]
Does something like:

CROW_ROUTE(app, "/add_json")
      .methods(crow::HTTPMethod::Option)([](const crow::request& req) {
// here you can set the response code 
      });

not work?

**Comment by gittiver** [comment]
will be closed as no answer to comment.

**Comment by avaliente-evs** [comment]
Sorry did not have time to investigate more.
On the website: https://crowcpp.org/master/guides/routes/ it says 

> Crow handles OPTIONS method automatically. The HEAD method is handled automatically unless defined in a route. Adding OPTIONS to a route's methods has no effect.

I did not try exactly the example you posted here, but I did try in a dynamic route and the handler was not called

**Comment by gittiver** [comment]
It seems that the http Standard prefers 200 but also allows 204. (see https://www.rfc-editor.org/rfc/rfc9110.html#name-200-ok)

Currently 204 is hardcoded in routing.h.

As a workaround you could patch this.

As it seems that some browsers depend on it (see https://stackoverflow.com/questions/14675330/what-http-response-code-should-be-used-for-an-option-request) I would propose to add an Cmake Option "HTTP_OPTION_RESPONSE_RETURNS_OK" which defaults to off and to switch the return code based on that option.

Alternatively we could switch in general to 200 but may run into issues maybe with older browsers or clients which make special assumptions.

**Comment by MoxueKulin** [comment]
Can the routing. h file be modified to return the Access Control Allow Origin request header for OPTIONS pre check requests?

**Comment by gittiver** [comment]
Yes, that modification could be done. But normally it should be supported by CORS Middleware.
see https://github.com/CrowCpp/Crow/blob/master/examples/middlewares/example_cors.cpp and https://crowcpp.org/master/guides/included-middleware/. 
If that setup does not work, feel free to come back.
